<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>817</width>
    <height>818</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QTabWidget" name="TAB">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>821</width>
      <height>771</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>2</number>
    </property>
    <widget class="QWidget" name="DBC_Writer">
     <attribute name="title">
      <string>DBC Writer</string>
     </attribute>
     <widget class="QLineEdit" name="sg_endian">
      <property name="geometry">
       <rect>
        <x>50</x>
        <y>270</y>
        <width>131</width>
        <height>22</height>
       </rect>
      </property>
     </widget>
     <widget class="QLineEdit" name="sg_type">
      <property name="geometry">
       <rect>
        <x>190</x>
        <y>270</y>
        <width>131</width>
        <height>22</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label_15">
      <property name="geometry">
       <rect>
        <x>190</x>
        <y>250</y>
        <width>48</width>
        <height>21</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>12</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Type</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_3">
      <property name="geometry">
       <rect>
        <x>50</x>
        <y>250</y>
        <width>48</width>
        <height>16</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>12</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Endian</string>
      </property>
     </widget>
     <widget class="QTextBrowser" name="DBC_temp">
      <property name="geometry">
       <rect>
        <x>50</x>
        <y>370</y>
        <width>521</width>
        <height>291</height>
       </rect>
      </property>
     </widget>
     <widget class="QPushButton" name="Signal">
      <property name="geometry">
       <rect>
        <x>600</x>
        <y>420</y>
        <width>161</width>
        <height>24</height>
       </rect>
      </property>
      <property name="text">
       <string>Signal syntax 적용</string>
      </property>
     </widget>
     <widget class="QPushButton" name="Save">
      <property name="geometry">
       <rect>
        <x>600</x>
        <y>500</y>
        <width>161</width>
        <height>24</height>
       </rect>
      </property>
      <property name="text">
       <string>Save As...</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_14">
      <property name="geometry">
       <rect>
        <x>230</x>
        <y>-120</y>
        <width>181</width>
        <height>31</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>16</pointsize>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>DBC File Creator</string>
      </property>
     </widget>
     <widget class="QPushButton" name="Message">
      <property name="geometry">
       <rect>
        <x>600</x>
        <y>380</y>
        <width>161</width>
        <height>24</height>
       </rect>
      </property>
      <property name="text">
       <string>Message syntax 적용</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_28">
      <property name="geometry">
       <rect>
        <x>470</x>
        <y>130</y>
        <width>211</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string/>
      </property>
     </widget>
     <widget class="QLabel" name="label_29">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>320</y>
        <width>128</width>
        <height>28</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>16</pointsize>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>DBC Preview</string>
      </property>
     </widget>
     <widget class="QLabel" name="label">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>40</y>
        <width>150</width>
        <height>28</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>16</pointsize>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>CAN Messages</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="CAN_id">
      <property name="geometry">
       <rect>
        <x>50</x>
        <y>100</y>
        <width>171</width>
        <height>22</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="can">
      <property name="geometry">
       <rect>
        <x>50</x>
        <y>80</y>
        <width>54</width>
        <height>21</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>12</pointsize>
       </font>
      </property>
      <property name="text">
       <string>CAN ID</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="Comment_MSG">
      <property name="geometry">
       <rect>
        <x>510</x>
        <y>100</y>
        <width>251</width>
        <height>22</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label_30">
      <property name="geometry">
       <rect>
        <x>510</x>
        <y>80</y>
        <width>72</width>
        <height>21</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>12</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Comment</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_31">
      <property name="geometry">
       <rect>
        <x>330</x>
        <y>250</y>
        <width>72</width>
        <height>21</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>12</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Comment</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="Comment_SG">
      <property name="geometry">
       <rect>
        <x>330</x>
        <y>270</y>
        <width>431</width>
        <height>22</height>
       </rect>
      </property>
     </widget>
     <widget class="QPushButton" name="Delete_Line">
      <property name="geometry">
       <rect>
        <x>600</x>
        <y>460</y>
        <width>161</width>
        <height>24</height>
       </rect>
      </property>
      <property name="text">
       <string>Delete Line</string>
      </property>
     </widget>
     <widget class="QPushButton" name="Open_File_With">
      <property name="geometry">
       <rect>
        <x>50</x>
        <y>670</y>
        <width>121</width>
        <height>24</height>
       </rect>
      </property>
      <property name="text">
       <string>Open File With</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="File_Path">
      <property name="geometry">
       <rect>
        <x>180</x>
        <y>670</y>
        <width>391</width>
        <height>22</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label_12">
      <property name="geometry">
       <rect>
        <x>540</x>
        <y>190</y>
        <width>30</width>
        <height>21</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>12</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Max</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="sg_len">
      <property name="geometry">
       <rect>
        <x>260</x>
        <y>210</y>
        <width>61</width>
        <height>22</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label_10">
      <property name="geometry">
       <rect>
        <x>400</x>
        <y>190</y>
        <width>44</width>
        <height>21</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>12</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Offset</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_9">
      <property name="geometry">
       <rect>
        <x>330</x>
        <y>190</y>
        <width>38</width>
        <height>21</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>12</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Scale</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_13">
      <property name="geometry">
       <rect>
        <x>620</x>
        <y>190</y>
        <width>30</width>
        <height>21</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>12</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Unit</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="sg_name">
      <property name="geometry">
       <rect>
        <x>50</x>
        <y>210</y>
        <width>133</width>
        <height>22</height>
       </rect>
      </property>
     </widget>
     <widget class="QLineEdit" name="sg_start">
      <property name="geometry">
       <rect>
        <x>190</x>
        <y>210</y>
        <width>61</width>
        <height>22</height>
       </rect>
      </property>
     </widget>
     <widget class="QLineEdit" name="sg_max">
      <property name="geometry">
       <rect>
        <x>540</x>
        <y>210</y>
        <width>61</width>
        <height>22</height>
       </rect>
      </property>
     </widget>
     <widget class="QLineEdit" name="sg_offset">
      <property name="geometry">
       <rect>
        <x>400</x>
        <y>210</y>
        <width>61</width>
        <height>22</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label_2">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>150</y>
        <width>123</width>
        <height>28</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>16</pointsize>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>CAN Signals</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="msg_len">
      <property name="geometry">
       <rect>
        <x>370</x>
        <y>100</y>
        <width>133</width>
        <height>22</height>
       </rect>
      </property>
     </widget>
     <widget class="QLineEdit" name="sg_min">
      <property name="geometry">
       <rect>
        <x>470</x>
        <y>210</y>
        <width>61</width>
        <height>22</height>
       </rect>
      </property>
     </widget>
     <widget class="QLineEdit" name="msg_name">
      <property name="geometry">
       <rect>
        <x>230</x>
        <y>100</y>
        <width>133</width>
        <height>22</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label_6">
      <property name="geometry">
       <rect>
        <x>50</x>
        <y>190</y>
        <width>43</width>
        <height>21</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>12</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Name</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_7">
      <property name="geometry">
       <rect>
        <x>190</x>
        <y>190</y>
        <width>35</width>
        <height>21</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>12</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Start</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="sg_unit">
      <property name="geometry">
       <rect>
        <x>620</x>
        <y>210</y>
        <width>141</width>
        <height>22</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label_8">
      <property name="geometry">
       <rect>
        <x>260</x>
        <y>190</y>
        <width>51</width>
        <height>21</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>12</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Length</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_5">
      <property name="geometry">
       <rect>
        <x>370</x>
        <y>80</y>
        <width>51</width>
        <height>21</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>12</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Length</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="sg_factor">
      <property name="geometry">
       <rect>
        <x>330</x>
        <y>210</y>
        <width>61</width>
        <height>22</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label_11">
      <property name="geometry">
       <rect>
        <x>470</x>
        <y>190</y>
        <width>28</width>
        <height>21</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>12</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Min</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_4">
      <property name="geometry">
       <rect>
        <x>230</x>
        <y>80</y>
        <width>43</width>
        <height>21</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>12</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Name</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="DBC_Reader">
     <attribute name="title">
      <string>DBC Reader</string>
     </attribute>
     <widget class="QLineEdit" name="ReadFIle_Path">
      <property name="geometry">
       <rect>
        <x>160</x>
        <y>20</y>
        <width>621</width>
        <height>22</height>
       </rect>
      </property>
     </widget>
     <widget class="QPushButton" name="Open_ReadFiles">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>20</y>
        <width>111</width>
        <height>24</height>
       </rect>
      </property>
      <property name="text">
       <string>Open FIle With</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_33">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>50</y>
        <width>142</width>
        <height>27</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>15</pointsize>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>CAN Messages</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_34">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>290</y>
        <width>116</width>
        <height>27</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>15</pointsize>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>CAN Signals</string>
      </property>
     </widget>
     <widget class="QComboBox" name="select">
      <property name="geometry">
       <rect>
        <x>170</x>
        <y>290</y>
        <width>101</width>
        <height>22</height>
       </rect>
      </property>
     </widget>
     <widget class="QTableWidget" name="Can_Message_Table">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>80</y>
        <width>751</width>
        <height>192</height>
       </rect>
      </property>
     </widget>
     <widget class="QTableWidget" name="CAN_Signal_Table">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>320</y>
        <width>751</width>
        <height>211</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label_16">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>550</y>
        <width>131</width>
        <height>27</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>15</pointsize>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>DBC File View</string>
      </property>
     </widget>
     <widget class="QTextBrowser" name="DBC_Read_Temp">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>580</y>
        <width>751</width>
        <height>141</height>
       </rect>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tab">
     <attribute name="title">
      <string>CAN Data Viewer</string>
     </attribute>
     <widget class="QTableWidget" name="CAN_Data_Table">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>620</y>
        <width>591</width>
        <height>101</height>
       </rect>
      </property>
      <attribute name="horizontalHeaderDefaultSectionSize">
       <number>90</number>
      </attribute>
     </widget>
     <widget class="QLabel" name="label_17">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>30</y>
        <width>166</width>
        <height>27</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>15</pointsize>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>CAN Data Viewer</string>
      </property>
     </widget>
     <widget class="QComboBox" name="Select2">
      <property name="geometry">
       <rect>
        <x>520</x>
        <y>70</y>
        <width>101</width>
        <height>22</height>
       </rect>
      </property>
     </widget>
     <widget class="QTextBrowser" name="Data_Result">
      <property name="geometry">
       <rect>
        <x>380</x>
        <y>100</y>
        <width>411</width>
        <height>501</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label_18">
      <property name="geometry">
       <rect>
        <x>50</x>
        <y>116</y>
        <width>90</width>
        <height>27</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>15</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Throttling</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_19">
      <property name="geometry">
       <rect>
        <x>70</x>
        <y>230</y>
        <width>41</width>
        <height>27</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>15</pointsize>
       </font>
      </property>
      <property name="text">
       <string>RPM</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_20">
      <property name="geometry">
       <rect>
        <x>70</x>
        <y>350</y>
        <width>50</width>
        <height>27</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>15</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Km/h</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_21">
      <property name="geometry">
       <rect>
        <x>40</x>
        <y>460</y>
        <width>98</width>
        <height>27</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>15</pointsize>
       </font>
      </property>
      <property name="text">
       <string>ENG_Temp</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_22">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>560</y>
        <width>138</width>
        <height>27</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>15</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Steering_Angle</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_23">
      <property name="geometry">
       <rect>
        <x>380</x>
        <y>70</y>
        <width>128</width>
        <height>27</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>15</pointsize>
       </font>
      </property>
      <property name="text">
       <string>CAN Message</string>
      </property>
     </widget>
     <widget class="QLCDNumber" name="Throttling">
      <property name="geometry">
       <rect>
        <x>180</x>
        <y>100</y>
        <width>161</width>
        <height>61</height>
       </rect>
      </property>
     </widget>
     <widget class="QLCDNumber" name="Steering_Angle">
      <property name="geometry">
       <rect>
        <x>180</x>
        <y>540</y>
        <width>161</width>
        <height>61</height>
       </rect>
      </property>
     </widget>
     <widget class="QLCDNumber" name="ENG_Temp">
      <property name="geometry">
       <rect>
        <x>180</x>
        <y>440</y>
        <width>161</width>
        <height>61</height>
       </rect>
      </property>
     </widget>
     <widget class="QLCDNumber" name="Velocity">
      <property name="geometry">
       <rect>
        <x>180</x>
        <y>330</y>
        <width>161</width>
        <height>61</height>
       </rect>
      </property>
     </widget>
     <widget class="QLCDNumber" name="RPM">
      <property name="geometry">
       <rect>
        <x>180</x>
        <y>210</y>
        <width>161</width>
        <height>61</height>
       </rect>
      </property>
     </widget>
     <widget class="QPushButton" name="Graph">
      <property name="geometry">
       <rect>
        <x>650</x>
        <y>620</y>
        <width>135</width>
        <height>101</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>12</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Graph,
HIstogram</string>
      </property>
     </widget>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>817</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
   </widget>
   <widget class="QMenu" name="menuEdit">
    <property name="title">
     <string>Edit</string>
    </property>
   </widget>
   <widget class="QMenu" name="menuView">
    <property name="title">
     <string>View</string>
    </property>
   </widget>
   <widget class="QMenu" name="menuDebug">
    <property name="title">
     <string>Debug</string>
    </property>
   </widget>
   <widget class="QMenu" name="menuAnalyze">
    <property name="title">
     <string>Analyze</string>
    </property>
   </widget>
   <widget class="QMenu" name="menuTools">
    <property name="title">
     <string>Tools</string>
    </property>
   </widget>
   <widget class="QMenu" name="menuWinodows">
    <property name="title">
     <string>Winodows</string>
    </property>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuEdit"/>
   <addaction name="menuView"/>
   <addaction name="menuDebug"/>
   <addaction name="menuAnalyze"/>
   <addaction name="menuTools"/>
   <addaction name="menuWinodows"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
